### PR TITLE
Overrided unhandledFilters()

### PR DESCRIPTION
**Pull request by weiqingy**
Changes:

This PR is to use Spark API [unhandledFilters()](https://github.com/apache/spark/blob/511f52f8423e151b0d0133baf040d34a0af3d422/sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala#L226) to tell Spark about filters we aren't implementing as opposed to returning all the filters. Currently, in SHC, the following filters are handled by HBase:

- EqualTo (singleKey, SingleColumnValueFilter)
- StringStartsWith
- StringEndsWith
- StringContains
- Not(StringContains)
- In

So, with this PR, the filters above will not be handled by Spark again.

### CODE REVIEW

**Comment by weiqingy** [comment]
Changes:

This PR is to use Spark API [unhandledFilters()](https://github.com/apache/spark/blob/511f52f8423e151b0d0133baf040d34a0af3d422/sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala#L226) to tell Spark about filters we aren't implementing as opposed to returning all the filters. Currently, in SHC, the following filters are handled by HBase:

- EqualTo (singleKey, SingleColumnValueFilter)
- StringStartsWith
- StringEndsWith
- StringContains
- Not(StringContains)
- In

So, with this PR, the filters above will not be handled by Spark again.

**Comment by weiqingy** [comment]
Hi, @dongjoon-hyun Could you please help to review this PR? Thanks.

**Comment by dongjoon-hyun** [comment]
This might have conflict with #52 . Which one do you want to proceed first?

**Comment by weiqingy** [comment]
Hi, @dongjoon-hyun, I have removed the conflicts.

**Comment by dongjoon-hyun** [comment]
I see. BTW, we don't use Travis?

**Comment by dongjoon-hyun** [comment]
At every review, we need a local build. :(

**Comment by weiqingy** [comment]
@dongjoon-hyun Yes, SHC will add Travis.

File: core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala [hunk 1]
```diff
@@ -224,6 +224,11 @@ case class HBaseRelation(
 
   override val schema: StructType = userSpecifiedschema.getOrElse(catalog.toDataType)
 
+  // Check if filters can be handled in HBase side
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala]: This function is a pruning function, not a checking function.

Comment (weiqingy) [inline core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala]: I have updated the PR to resolved this.

**Comment by weiqingy** [comment]
Thanks for the review. @dongjoon-hyun

**Pull request merged**